src/components/ContextualSaveBar/ContextualSaveBar.tsx

```diff
@@ -1,32 +1,36 @@
 import * as React from 'react';
+import compose from '@shopify/react-compose';
 import isEqual from 'lodash/isEqual';
-import {
-  ContextualSaveBarProps,
-  FrameContext,
-  frameContextTypes,
-} from '../Frame';
+import {ContextualSaveBarProps, FrameContext, Consumer} from '../Frame';
+import withContext from '../WithContext';
+import {WithContextTypes} from '../../types';
+import {withAppProvider, WithAppProviderProps} from '../AppProvider';
 
 // The script in the styleguide that generates the Props Explorer data expects
 // a component's props to be found in the Props interface. This silly workaround
 // ensures that the Props Explorer table is generated correctly, instead of
 // crashing if we write `ContextualSaveBar extends React.Component<ContextualSaveBarProps>`
 interface Props extends ContextualSaveBarProps {}
+export type ComposedProps = Props &
+  WithAppProviderProps &
+  WithContextTypes<FrameContext>;
 
-class ContextualSaveBar extends React.PureComponent<Props, never> {
-  static contextTypes = frameContextTypes;
-  context: FrameContext;
-
+class ContextualSaveBar extends React.PureComponent<ComposedProps, never> {
   componentDidMount() {
-    this.context.frame.setContextualSaveBar(this.props);
+    const {props} = this;
+    const {polaris, context, ...rest} = props;
+    props.context.frame.setContextualSaveBar(rest);
   }
 
   componentWillUnmount() {
-    this.context.frame.removeContextualSaveBar();
+    this.props.context.frame.removeContextualSaveBar();
   }
 
-  componentDidUpdate(oldProps: Props) {
-    if (contextualSaveBarHasChanged(this.props, oldProps)) {
-      this.context.frame.setContextualSaveBar(this.props);
+  componentDidUpdate(oldProps: ComposedProps) {
+    const {props} = this;
+    const {polaris, context, ...rest} = props;
```

Comment:
This bit feels wrong to me. I don't think I should need to pull `polaris` and `context` off `props` in order for this to work.

Reply:
Technically it'll work without deconstruction since we don't normally spread props. I'm surprised we didn't have a TS error before, since `polaris` always existed on props. I wonder if we'll still have one now? 

This is just a by product of the injector pattern. Since our smart components will always receive additional props through `HoC`s that won't be passed to the component upon usage.